File: sdk/runanywhere-swift/Sources/RunAnywhere/Public/Configuration/ModelProviderConfig.swift
import Foundation

/// Model provider configuration
public struct ModelProviderConfig {
    /// Provider name (e.g., "HuggingFace", "Kaggle")
    public let provider: String

    /// Authentication credentials
    public let credentials: ProviderCredentials?

    /// Whether this provider is enabled
    public let enabled: Bool

    public init(
        provider: String,
        credentials: ProviderCredentials? = nil,
        enabled: Bool = true
    ) {
        self.provider = provider
        self.credentials = credentials
        self.enabled = enabled
    }
}

File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/configuration/ModelProviderConfig.kt
package com.runanywhere.sdk.configuration

/**
 * Model provider configuration
 */
data class ModelProviderConfig(
    /**
     * Provider name (e.g., "HuggingFace", "Kaggle")
     */
    val provider: String,
    
    /**
     * Authentication credentials
     */
    val credentials: ProviderCredentials? = null,
    
    /**
     * Whether this provider is enabled
     */
    val enabled: Boolean = true
)

/**
 * Provider credentials
 */
data class ProviderCredentials(
    val apiKey: String? = null,
    val username: String? = null,
    val password: String? = null,
    val token: String? = null
) 